dvc/tree/http.py

```diff
@@ -194,7 +195,7 @@ def chunks():
                             break
                         yield chunk
 
-        response = self.request("POST", to_info.url, data=chunks())
+        response = self.request(self.http_method, to_info.url, data=chunks())
```

Comment:
Hi @rfdickerson !

I suppose you need to use PUT in your scenario, right? Could you elaborate on what your scenario is?

Reply:
Absolutely! Our company at Expedia group uses JFrog's Artifactory for storing binary artifacts. While Artifactory supports an HTTP/HTTPS interface, instead of POST it only accepts PUT for uploading new artifacts. This PR provides an alternative in this case.

Reply:
PUT seems like a more appropriate method than POST, considering that we already know the URL of cache file that we are uploading. I don't have a good statistics on how often people use one or another, so not sure if maybe we should consider switching to PUT by default along the way. We'll keep an eye on it, maybe we'll switch the defaults in the future.

Let's keep it as is for now. Thank you so much!